Mathlib/LinearAlgebra/TensorProduct/Basic.lean

```diff
@@ -497,99 +105,14 @@ variable [Module R P]
-the given bilinear map `M → N → P`. -/
-def lift : M ⊗[R] N →ₗ[R] P :=
-  { liftAux f with map_smul' := liftAux.smul }
-
-variable {f}
-
-@[simp]
-theorem lift.tmul (x y) : lift f (x ⊗ₜ y) = f x y :=
-  rfl
-
-@[simp]
-theorem lift.tmul' (x y) : (lift f).1 (x ⊗ₜ y) = f x y :=
-  rfl
-
-theorem ext' {g h : M ⊗[R] N →ₗ[R] P} (H : ∀ x y, g (x ⊗ₜ y) = h (x ⊗ₜ y)) : g = h :=
-  LinearMap.ext fun z =>
-    TensorProduct.induction_on z (by simp_rw [LinearMap.map_zero]) H fun x y ihx ihy => by
-      rw [g.map_add, h.map_add, ihx, ihy]
-
-theorem lift.unique {g : M ⊗[R] N →ₗ[R] P} (H : ∀ x y, g (x ⊗ₜ y) = f x y) : g = lift f :=
-  ext' fun m n => by rw [H, lift.tmul]
-
-theorem lift_mk : lift (mk R M N) = LinearMap.id :=
-  Eq.symm <| lift.unique fun _ _ => rfl
-
-theorem lift_compr₂ (g : P →ₗ[R] Q) : lift (f.compr₂ g) = g.comp (lift f) :=
-  Eq.symm <| lift.unique fun _ _ => by simp
-
-theorem lift_mk_compr₂ (f : M ⊗ N →ₗ[R] P) : lift ((mk R M N).compr₂ f) = f := by
-  rw [lift_compr₂ f, lift_mk, LinearMap.comp_id]
-
-/-- This used to be an `@[ext]` lemma, but it fails very slowly when the `ext` tactic tries to apply
-it in some cases, notably when one wants to show equality of two linear maps. The `@[ext]`
-attribute is now added locally where it is needed. Using this as the `@[ext]` lemma instead of
-`TensorProduct.ext'` allows `ext` to apply lemmas specific to `M →ₗ _` and `N →ₗ _`.
-
-See note [partially-applied ext lemmas]. -/
-theorem ext {g h : M ⊗ N →ₗ[R] P} (H : (mk R M N).compr₂ g = (mk R M N).compr₂ h) : g = h := by
-  rw [← lift_mk_compr₂ g, H, lift_mk_compr₂]
+variable {f : M →ₗ[R] N →ₗ[R] P}
 
 attribute [local ext high] ext
 
 example : M → N → (M → N → P) → P := fun m => flip fun f => f m
 
 variable (R M N P)
 
-/-- Linearly constructing a linear map `M ⊗ N → P` given a bilinear map `M → N → P`
-with the property that its composition with the canonical bilinear map `M → N → M ⊗ N` is
-the given bilinear map `M → N → P`. -/
-def uncurry : (M →ₗ[R] N →ₗ[R] P) →ₗ[R] M ⊗[R] N →ₗ[R] P :=
-  LinearMap.flip <| lift <| LinearMap.lflip.comp (LinearMap.flip LinearMap.id)
-
-variable {R M N P}
-
-@[simp]
-theorem uncurry_apply (f : M →ₗ[R] N →ₗ[R] P) (m : M) (n : N) :
-    uncurry R M N P f (m ⊗ₜ n) = f m n := by rw [uncurry, LinearMap.flip_apply, lift.tmul]; rfl
-
-variable (R M N P)
-
-/-- A linear equivalence constructing a linear map `M ⊗ N → P` given a bilinear map `M → N → P`
-with the property that its composition with the canonical bilinear map `M → N → M ⊗ N` is
-the given bilinear map `M → N → P`. -/
-def lift.equiv : (M →ₗ[R] N →ₗ[R] P) ≃ₗ[R] M ⊗[R] N →ₗ[R] P :=
-  { uncurry R M N P with
-    invFun := fun f => (mk R M N).compr₂ f
-    left_inv := fun _ => LinearMap.ext₂ fun _ _ => lift.tmul _ _
-    right_inv := fun _ => ext' fun _ _ => lift.tmul _ _ }
-
 @[simp]
 theorem lift.equiv_apply (f : M →ₗ[R] N →ₗ[R] P) (m : M) (n : N) :
     lift.equiv R M N P f (m ⊗ₜ n) = f m n :=
```

Comment:
Similarly, I would vote that things about the universal property itself are "primitive", and so belong in `defs`.